extensions/Worker.Extensions.SignalRService/src/SignalRServiceDependencyInjectionExtensions.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.Functions.Worker.SignalRService;
+using Microsoft.Azure.SignalR.Management;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class SignalRServiceDependencyInjectionExtensions
+    {
+        public static IServiceCollection AddServerlessHub<THub>(this IServiceCollection services)
+            where THub : ServerlessHub
+            => services.AddServerlessHub(typeof(THub));
+
+        public static IServiceCollection AddServerlessHub<THub>(this IServiceCollection services, Action<ServiceManagerBuilder> configure)
+            where THub : ServerlessHub
+            => services.AddServerlessHub(typeof(THub), configure);
+
+        public static IServiceCollection AddServerlessHub(this IServiceCollection services, Type hubType)
+           => services.AddServerlessHub(hubType, _ => { });
+
+        public static IServiceCollection AddServerlessHub(this IServiceCollection services, Type hubType, Action<ServiceManagerBuilder> configure)
+        {
+            var allBaseTypes = hubType.AllBaseTypes().ToArray();
```

Comment:
I like to use this API for runtime type checks:

``` CSharp
typeof(ServerlessHub).IsAssignableFrom(hubType)
```

This will check that `hubType` derives from `ServerlessHub`

Reply:
Updated